src/workflows-app/RunDetails.test.js

```diff
@@ -242,6 +242,29 @@ describe('BaseRunDetails - render smoke test', () => {
     });
   });
 
+  it('shows expected cost details', async () => {
```

Comment:
These rendering tests are fine, but there are many cases that the `getTaskCost` and `calculateTotalCost` functions handle that could be unit-tested. That would also give us an easier framework to validate edge cases without having to wrap everything into a full page-render test.